### PR TITLE
DOC-470 Debug bundle in Redpanda Console

### PR DESCRIPTION
**Pull request by JakeSCahill**
## Description

This PR adds docs for the new feature to generate debug bundles in Redpanda Console. Additionally, it introduces a unified 'Troubleshoot' section that centralizes content on cluster diagnostics, error resolution, and debug bundle generation across environments.

Previously, the Cluster Diagnostics documentation covered self-tests but lacked information on Kubernetes. Similarly, only the Troubleshooting document for Kubernetes included error resolution content. This update consolidates all troubleshooting content (in `troubleshoot/partials`) to ensure both Linux and Kubernetes users have consistent access to diagnostics and error resolution guidance.

This single-source approach simplifies maintenance and provides users with cohesive troubleshooting instructions regardless of environment.

Fixes:

- https://redpandadata.atlassian.net/browse/DOC-562
- https://redpandadata.atlassian.net/browse/DOC-507
- https://redpandadata.atlassian.net/browse/DOC-470
- https://redpandadata.atlassian.net/browse/DOC-425

Relies on https://github.com/redpanda-data/seastar/pull/152 and an RC4 release to work in Docker Desktop

Review deadline: 5 November 

## Page previews

- [Manage Debug Bundles in Redpanda Console](https://deploy-preview-825--redpanda-docs-preview.netlify.app/24.3/console/ui/generate-bundle/)
- [New 'Troubleshoot' section](https://deploy-preview-825--redpanda-docs-preview.netlify.app/24.3/troubleshoot/)

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/console/pages/ui/generate-bundle.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+== Generate a debug bundle
+
+You can generate a debug bundle for all brokers in the cluster using Redpanda Console. After generating the bundle, download it onto your local computer for inspection.
+
+. Log in as an admin user.
+. Click *Admin* in the sidebar.
++
+This option is visible only when you are logged in as an admin user.
+. Go to *Debug bundle*.
+. Click *Generate new*, then click *Generate default*.
```
Comment (JakeSCahill) [inline modules/console/pages/ui/generate-bundle.adoc]:
```suggestion
. Click *Generate new*.
```

File: modules/console/pages/ui/generate-bundle.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+. Wait until the process is complete.
+. Click *debug-bundle.zip* to download the bundle on your local computer.
+. Unzip the file to inspect the contents.
+
+== Delete a debug bundle
+
+Each broker stores only one debug bundle at a time. To generate a new debug bundle, you must delete the existing one.
```
Comment (JakeSCahill) [inline modules/console/pages/ui/generate-bundle.adoc]:
```suggestion
To generate a new debug bundle, you must delete the existing one.
```

File: modules/console/pages/ui/generate-bundle.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
+= Manage Debug Bundles in Redpanda Console
+:description: Learn how to generate, download, and delete debug bundles in Redpanda Console for comprehensive cluster diagnostics.
+
+{description}
+
+== Prerequisites
+
+- You must have an Enterprise Edition license. https://www.redpanda.com/contact[Request a license^] if you don't have one already.
```
Comment (kbatuigas) [inline modules/console/pages/ui/generate-bundle.adoc]: I think we've lately been using a partial (wrapped in a NOTE) to point out the licensing requirement. Should we consider doing that here?

File: modules/troubleshoot/partials/cluster-diagnostics.adoc [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+kubectl get redpandas.cluster.redpanda.com -o yaml --namespace <namespace>
+----
+
+In both the HelmRelease and the Redpanda resource, the condition section reveals the ongoing status of the Helm installation. These conditions provide information on the success, failure, or pending status of various operations.
+endif::[]
+
+[[self-test]]
+== Self-test benchmarks
+
+When anomalous behavior arises in a cluster, you can determine if it's caused by issues with hardware, such as disk drives or network interfaces (NICs) by running xref:reference:rpk/rpk-cluster/rpk-cluster-self-test.adoc[rpk cluster self-test] (self-test) to assess their performance and compare it to vendor specifications.
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/cluster-diagnostics.adoc]: Curious why we repeat "self-test" in parentheses. Maybe we could style `rpk cluster self-test` to better distinguish?

File: modules/troubleshoot/partials/cluster-diagnostics.adoc [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+The `rpk cluster self-test` command runs a set of benchmarks to gauge the maximum performance of a machine's disks and network connections:
+
+* **Disk tests**: Measures throughput and latency by performing concurrent sequential operations.
+* **Network tests**: Selects unique pairs of Redpanda brokers as client/server pairs and runs throughput tests between them.
+
+Each benchmark runs for a configurable duration and returns IOPS, throughput, and latency metrics. This helps you determine if hardware performance aligns with expected vendor specifications.
+
+=== Cloud storage tests
+
+To confirm your cloud storage is configured correctly for Tiered Storage, use the self-test command.
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/cluster-diagnostics.adoc]:
```suggestion
You can also use the self-test command to confirm your cloud storage is configured correctly for xref:manage:tiered-storage.adoc[Tiered Storage].
```
Bit of a paraphrase for a smoother introduction to Tiered Storage in this doc

File: modules/troubleshoot/partials/cluster-diagnostics.adoc [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+--
+======
+
+After applying these changes, verify the log level by <<view-redpanda-logs, checking the initial output of the logs>> for the Redpanda Pods.
+
+=== Override the default log level for Redpanda subsystems
+
+You can override the log levels for individual subsystems, such as `rpc` and `kafka`, for more detailed logging control. Overrides exist for the entire length of the running Redpanda process.
+
+TIP: To temporarily override the log level for individual subsystems, you can use the xref:reference:rpk/rpk-redpanda/rpk-redpanda-admin-config-log-level-set.adoc[`rpk-redpanda-admin-config-log-level-set`] command.
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/cluster-diagnostics.adoc]:
```suggestion
TIP: To temporarily override the log level for individual subsystems, you can use the xref:reference:rpk/rpk-redpanda/rpk-redpanda-admin-config-log-level-set.adoc[`rpk redpanda admin config log-level set`] command.
```
For consistency

File: modules/troubleshoot/partials/errors-and-solutions.adoc [hunk 7]
```diff
@@ -719,11 +491,105 @@ Check the following:
...
 
+If you're using rpk, ensure to add the `-X tls.enabled` flag, and any other necessary TLS flags such as the TLS certificate:
+
+[,bash]
+----
+ifdef::env-kubernetes[kubectl exec <pod-name> -c redpanda --namespace <namespace> -- \]
+rpk cluster info -X tls.enabled=true
+----
+
+For all available flags, see the xref:reference:rpk/index.adoc[rpk command reference].
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/errors-and-solutions.adoc]: Should we link to reference:rpk/rpk-x-options/ instead?

File: modules/troubleshoot/partials/errors-and-solutions.adoc [hunk 8]
```diff
@@ -719,11 +491,105 @@ Check the following:
...
+This error appears when a cluster has TLS enabled, and you try to access the admin API without passing the required TLS parameters.
+
+[.no-copy]
+----
+Retrying POST for error: Post "http://127.0.0.1:9644/v1/security/users": net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x15\x03\x03\x00\x02\x02"
+----
+
+If you're using rpk, ensure to include the TLS flags.
+
+For all available flags, see the xref:reference:rpk/index.adoc[rpk command reference].
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/errors-and-solutions.adoc]: Same comment as above re: rpk -X

File: modules/troubleshoot/partials/errors-and-solutions.adoc [hunk 9]
```diff
@@ -737,4 +603,18 @@ To fix this error, ensure that you use only one of the following methods to crea
...
+This error appears when you try to interact with a cluster that has SASL enabled without passing a user's credentials.
+
+[.no-copy]
+----
+unable to request metadata: broker closed the connection immediately after a request was issued, which happens when SASL is required but not provided: is SASL missing?
+----
+
+If you're using rpk, ensure to specify the `-X user`, `-X pass`, and `-X sasl.mechanism` flags.
+
+For all available flags, see the xref:reference:rpk/index.adoc[rpk command reference].
```
Comment (kbatuigas) [inline modules/troubleshoot/partials/errors-and-solutions.adoc]: Same comment as above re: rpk -X

**Review by kbatuigas — APPROVED** [review]
Just a few suggestions, I don't think there are any blockers.

Also, not sure if you meant to include all the test files and artifacts in this PR, so I ignored them. (I'm also not sure if the test stuff should have gone in a PR by itself?)

**Review by bojand — APPROVED** [review]

**Comment by JakeSCahill** [comment]
Tests were failing because we were waiting for a beta build of Console.

**Pull request merged**